python/cudf/cudf/core/index.py

```diff
@@ -298,7 +298,7 @@ def isnull(self):
         """
         from cudf.core.series import Series
 
-        return Series(self.as_column().isnull(), name=self.name)
+        return GenericIndex(self.as_column().isnull(), name=self.name)
```

Comment:
Instead of using `GenericIndex` here you should just use the column factory `as_index`